goodtables/pipeline/batch.py

```diff
@@ -19,6 +20,9 @@ class Batch(object):
     * `source_type`: 'csv' or 'datapackage'
     * `data_key`: The key for the data (only used when source_type is 'csv')
     * `schema_key`: The key for the schema (only used when source_type is 'csv').
+    * `batch_options`: Options to pass to each batch instance.
```

Comment:
All options are "batch_options" here, except the `pipeline_options` which is an object to pass to each pipeline. So, adding a generic `batch_options` does not follow the API pattern here. It would be better to have a keyword argument like `rate` or `frequency` or `sleep`, with a default value of `1`.

So, the kwarg would be (for example) `rate=1`. 

Then, the try/except for the rate is also a bit non-idiomatic. Once you set a sensible default, like `rate=1`, then you should do the checking in the `__init__` and not let bad values exist, only to be discovered in the `run` method.

So, something like this in `__init__`, to catch the error immediately when passed:

```

if not isinstance(rate, (int, float)):
    raise ValueError('Received non int or float for the rate argument.')
self.rate = rate
```